### PR TITLE
API for social login

### PR DESCRIPTION
**Pull request by shradhaN**
API endpoint: `{base_url}/v1/spree_oauth/social_login/:provider`
method: POST

Request Body:
```{
  "provider": "facebook",
  "uid": "3329220777170562",
  "info": {
    "email": "example@example.com",
    "name": "Example Example",
    "image": "http://graph.facebook.com/v4.0/3329220777170562/picture"
  },
  "credentials": {
    "token": "<token>",
    "expires_at": 1606145485,
    "expires": true
  },
  "extra": {
    "raw_info": {
      "email": "example@example.com",
      "name": "Example Example",
      "id": "3329220777170562"
    }
  }
}
```

Response Body:
```
{
    "access_token": "F7eVY71qJdIVVsZ-MmuZ1VXtSBGoI6Gsdoac3hrPAyg",
    "token_type": "Bearer",
    "expires_in": 7200,
    "refresh_token": "eGp3TPP4RyJD2jJG02KL-i7yl1Y90wbbpNSoYXyTcks",
    "created_at": 1601391609
}

### CODE REVIEW

File: app/controllers/spree/api/v1/omniauth_callbacks_controller.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
+
+  class Spree::Api::V1::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+    include Spree::Core::ControllerHelpers::Common
+    include Spree::Core::ControllerHelpers::Order
+    include Spree::Core::ControllerHelpers::Auth
+    include Spree::Core::ControllerHelpers::Store
```
Comment (poudelprakash) [inline app/controllers/spree/api/v1/omniauth_callbacks_controller.rb]: do we need all of these modules?

File: app/controllers/spree/api/v1/omniauth_callbacks_controller.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    skip_before_action :verify_authenticity_token
+
+    def login
+      eligible_providers = SpreeSocial::OAUTH_PROVIDERS.map { |p| p[1] if p[2] == 'true' }.compact
```
Comment (poudelprakash) [inline app/controllers/spree/api/v1/omniauth_callbacks_controller.rb]: isn't there any helper method for this available yet? does not look like it really belongs to login method

File: app/controllers/spree/api/v1/omniauth_callbacks_controller.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+      if !(eligible_providers.include?(auth_hash['provider']))
+        render json: {error: I18n.t('devise.omniauth_callbacks.provider_not_found', kind: auth_hash['provider'])}, status: :unprocessable_entity
```
Comment (poudelprakash) [inline app/controllers/spree/api/v1/omniauth_callbacks_controller.rb]: can we use failure_message method from https://github.com/heartcombo/devise/blob/master/app/controllers/devise/omniauth_callbacks_controller.rb ?

Comment (poudelprakash) [reply app/controllers/spree/api/v1/omniauth_callbacks_controller.rb]: we should be able to just raise error and base controller should be able to handle the exceptions..
we have lots of things going inside this action

File: config/routes.rb [hunk 4]
```diff
@@ -8,7 +8,14 @@
...
+      use_doorkeeper do
+        skip_controllers 'spree/omniauth_callbacks#login'
+      end
+      devise_scope :spree_user do
+        post '/spree_oauth/social_login/:provider', to: 'omniauth_callbacks#login'
+      end
+    end
   end
-end
+end
```
Comment (poudelprakash) [inline config/routes.rb]: last line missing

File: config/routes.rb [hunk 5]
```diff
@@ -8,7 +8,14 @@
 
   get 'account' => 'users#show', as: 'user_root'
 
-  namespace :admin do
-    resources :authentication_methods
+  namespace :api, defaults: { format: 'json' } do
```
Comment (poudelprakash) [inline config/routes.rb:15]: why did we remove admin routes?

I don't think we should be removing any routes..

File: app/controllers/spree/api/v1/omniauth_callbacks_controller.rb (hunk 2 above)
Comment (coolprobn) [reply app/controllers/spree/api/v1/omniauth_callbacks_controller.rb]: Would it make sense to use more sensible variable names instead of p?

**Comment by poudelprakash** [comment]
@shradhaN you have sensitive information in above description, 
please modify json to hide sensitive information..

We have checked this change and is working for us, you can create pr in spree-contrib's repo as well.
I am merging this for us :D

Thank you

**Comment by poudelprakash** [comment]
we will also need documentation in repo for this :D

**Pull request merged**